### PR TITLE
feature/GOL-1-add-unit-tests-buttons

### PR DESCRIPTION
**Pull request by GUIDOURI**
###  What was done?

- Created unit tests for all button variants (Filled, Elevated, Tonal, Outlined, Text) using React Testing Library.
- Covered edge cases and state changes (e.g., disabled, loading, with icons).
- Integrated animations and design tokens.

###  Why?

To ensure UI components follow expected behavior and appearance under different props and states, increasing maintainability and trust.

###  How to test?

   ````markdown
1. Run:
   ```bash
   pnpm test src/components/button
2. All button variants should be tested successfully.

3. Storybook renders should also match the tested components.

### CODE REVIEW

File: apps/ui/src/components/button/button.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+			{icon && <span className={styles.icon}>{icon}</span>}
+			<span>{children}</span>
+		</button>
+	);
+};
+
+export default Button;
+
+
+export const FilledButton = (properties: Omit<ButtonBaseProperties, 'variant'>) => (
```
Comment (ougarte) [inline apps/ui/src/components/button/button.tsx]: I recommend removing all variants `FilledButton | ElevatedButton | TonalButton  | OutlinedButton  | TextButton  | IconButton  | SegmentedButton`

File: apps/ui/src/components/button/index.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+	default as BaseButton,
+	default as ButtonTextComponent,
+	default as ButtonIconComponent,
+	default as ButtonFilledComponent,
+	default as ButtonElevatedComponent,
+	default as ButtonTonalComponent,
+	default as ButtonOutlinedComponent,
+	default as ButtonFABComponent,
+	default as ButtonSegmentedComponent,
+} from './button';
```
Comment (ougarte) [inline apps/ui/src/components/button/index.ts]: I recommend exporting a unique Button `export { default as Button } from './button'`

File: apps/ui/src/components/button/styles/animations.css [hunk 3]
```diff
@@ -0,0 +1,21 @@
+@import url('./variables.css');
+
+.button {
```
Comment (ougarte) [inline apps/ui/src/components/button/styles/animations.css]: I recommend moving this style to `'core.css'`

File: apps/ui/src/components/button/styles/button.module.css [hunk 4]
```diff
@@ -0,0 +1,94 @@
+@import './variables.css';
+@import './animations.css';
+
+/* Estilo base del botón */
+.buttonBase {
```
Comment (ougarte) [inline apps/ui/src/components/button/styles/button.module.css]: I recommend updating as `.button`

File: apps/ui/src/components/button/styles/button.module.css [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  overflow: hidden;
+}
+
+/* Modificador para el ancho completo */
+.fullWidth {
+  width: 100%;
+}
+
+/* Estilo para los íconos dentro del botón */
+.icon {
```
Comment (ougarte) [inline apps/ui/src/components/button/styles/button.module.css]: I recommend adding a namespace `.button .icon`

File: apps/ui/src/components/button/styles/button.module.css [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+  display: flex;
+  align-items: center;
+  justify-content: center;
+}
+
+/* Variantes de Material 3 */
+
+/* Estilo para el botón lleno */
+.filled {
```
Comment (ougarte) [inline apps/ui/src/components/button/styles/button.module.css]: I recommend moving into `.button` scope

```css
.button {
    &.filled {}
    /* add more styles here... */
}
```

File: apps/ui/src/components/button/button.tsx [hunk 7]
```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import clsx from 'clsx';
+import styles from './styles/button.module.css';
```
Comment (ougarte) [inline apps/ui/src/components/button/button.tsx]: I recommend removing `css modules`

File: apps/ui/src/components/button/styles/variants.css [hunk 8]
```diff
@@ -0,0 +1,21 @@
+@import url('./variables.css');
+
+.button {
+	/* By default is `text` button */
+
+	&.filled {
+		background-color: transparent;
```
Comment (ougarte) [inline apps/ui/src/components/button/styles/variants.css]: I recommend moving `background-color: transparent;` as 
```css
.button {
    background-color: transparent;
}
```

File: apps/ui/src/components/button/helpers.tsx [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import type { Properties } from './types';
+
+/**
+ *
+ * @param properties
+ * @returns
+ */
+// Después
```
Comment (ougarte) [inline apps/ui/src/components/button/helpers.tsx]: I recommend using English comments

File: apps/ui/src/components/button/__tests__/button.test.tsx [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+import {
+	FilledButton,
+	ElevatedButton,
+	TonalButton,
+	OutlinedButton,
+	TextButton,
+	IconButton,
+} from '../button';
+
+describe('Button Variants', () => {
```
Comment (ougarte) [inline apps/ui/src/components/button/__tests__/button.test.tsx]: I recommend organizing the unit test as follow:

```ts
describe('components/button', () => {
    describe('layout', () => {
        test('should render with default properties', () => {});
        test('should render filled default with default properties', () => {});
        test('should render elevated  default with default properties', () => {});
        // and so on...
    });
});
```

File: apps/ui/src/components/button/button.tsx [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+	variant = 'filled',
+	fullWidth = false,
+	icon,
+	selected = false,
+	className,
+	...properties
+}) => {
+	const classes = clsx(
+		'button',
+		`button--${variant}`,
```
Comment (ougarte) [inline apps/ui/src/components/button/button.tsx:34]: I recommend avoiding mixing CSS approaches... so that you could use as follow

```ts
clsx('button', variant, fullWidth, selected, className);
```

File: apps/ui/src/components/button/button.tsx [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+export interface ButtonBaseProperties extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  children: React.ReactNode;
+  variant?: ButtonVariant;
+  fullWidth?: boolean;
+  icon?: React.ReactNode;
+  selected?: boolean; 
+  className?: string;
+}
+
+const Button: React.FC<ButtonBaseProperties> = ({
```
Comment (ougarte) [inline apps/ui/src/components/button/button.tsx]: I recommend Funcional Components

```tsx
export default function Button(properties?: ButtonBaseProperties) {
    // code goes here...
}
```

File: apps/ui/src/components/button/__tests__/button.test.tsx [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+import '@testing-library/jest-dom';
+
+import {
+	FilledButton,
+	ElevatedButton,
+	TonalButton,
+	OutlinedButton,
+	TextButton,
+	IconButton,
+} from '../button';
```
Comment (ougarte) [inline apps/ui/src/components/button/__tests__/button.test.tsx]: Quick question: where are these buttons being exported?

File: apps/ui/src/components/button/button.tsx [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+	children,
+	variant = 'filled',
+	fullWidth = false,
+	icon,
+	selected = false,
+	className,
+	...properties
```
Comment (ougarte) [inline apps/ui/src/components/button/button.tsx]: I recommend wrapping `properties` with `toDefaults` helper

File: apps/ui/src/components/button/styles/button.module.css [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
+  border-radius: 0.5rem;
+  border: none;
+  cursor: pointer;
+  transition: background-color 0.3s ease, box-shadow 0.3s ease, transform 0.2s ease;
+  position: relative;
+  overflow: hidden;
+}
+
+
+.button--fullWidth {
```
Comment (ougarte) [inline apps/ui/src/components/button/styles/button.module.css]: I recommend moving to .button as follow

```css
.button {
    &.full-width { }
    .icon {}

    /* and more nested selectors */
}
```

File: apps/ui/src/components/button/styles/button.module.css
Comment (ougarte) [inline apps/ui/src/components/button/styles/button.module.css:1]: I recommend moving all styles to their respective `css` file, and then delete this `button.module.css` file

File: apps/ui/src/components/button/__tests__/button.test.tsx [hunk 16]
```diff
@@ -0,0 +1,40 @@
+import { render, screen } from '@testing-library/react';
+import Button from '../button';
+
+describe('Button', () => {
```
Comment (ougarte) [inline apps/ui/src/components/button/__tests__/button.test.tsx:4]: I recommend updating as `components/button`

File: apps/ui/src/components/button/__tests__/button.test.tsx [hunk 17]
```diff
@@ -0,0 +1,40 @@
...
+	it('renders with default variant (filled)', () => {
```
Comment (ougarte) [inline apps/ui/src/components/button/__tests__/button.test.tsx]: I recommend using `should` on **all unit-test** (ex: `should render by default`)

File: apps/ui/src/components/button/button.tsx [hunk 18]
```diff
@@ -0,0 +1,64 @@
...
+export type ButtonVariant = 'filled' | 'elevated' | 'tonal' | 'outlined' | 'text';
+
+export interface ButtonBaseProperties extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  children: React.ReactNode;
+  variant?: ButtonVariant;
+  fullWidth?: boolean;
+  icon?: React.ReactNode;
+  selected?: boolean;
+  className?: string;
+  layout?: 'horizontal' | 'vertical' | 'centered';
```
Comment (ougarte) [inline apps/ui/src/components/button/button.tsx]: I recommend changing `layout` by `direction` or `alignment`

File: apps/ui/src/components/button/button.tsx [hunk 19]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+const variantClassMap: Record<ButtonVariant, () => string> = {
```
Comment (ougarte) [inline apps/ui/src/components/button/button.tsx]: I think this variable is not necesary

File: apps/ui/src/components/button/button.tsx [hunk 20]
```diff
@@ -0,0 +1,64 @@
...
+
+	const classes = clsx(
+		'button',
+		variantClass,
+		fullWidth && 'button--fullWidth',
+		selected && 'button--selected',
+		layout && `layout--${layout}`,
+		'fade-in',
+		className
+	);
```
Comment (ougarte) [inline apps/ui/src/components/button/button.tsx:41]: I recommend changing as follow:

```tsx
clsx('button fade-in', variant, className, layout, {
    'button--fullWidth': fullWidth
    'button--selected': selected,
});
```

**Review by ougarte — APPROVED** [review]
